sdk/core/azure-core/src/main/java/com/azure/core/implementation/TypeUtil.java

```diff
@@ -3,16 +3,20 @@
 
 package com.azure.core.implementation;
 
+import java.lang.reflect.GenericDeclaration;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
 
 /**
  * Utility type exposing methods to deal with {@link Type}.
  */
 public final class TypeUtil {
+    private static final Map<Type, Type> SUPER_TYPE_MAP = new WeakHashMap<>();
```

Comment:
We want to use `WeakHashMap` to evict from the cache once the Type key becomes null/garbage collected right?

Reply:
I've changed it to be a HashMap.